server/sonar-server/src/main/java/org/sonar/server/batch/ProjectRepositoryLoader.java

```diff
@@ -88,8 +89,9 @@ public ProjectRepositories load(ProjectRepositoryQuery query) {
       if (module != null) {
         if (query.isPreview()) {
           // Scan permission is enough to analyze all projects but preview permission is limited to projects user can access
-          UserSession.get().checkComponentPermission(UserRole.USER, query.getModuleKey(),
-            "You're not authorized to access to project '" + module.name() + "', please contact your SonarQube administrator.");
+          if (!UserSession.get().hasProjectPermissionByUuid(UserRole.USER, module.projectUuid())) {
```

Comment:
Just to understand here, what was the issue with ``checkComponentPermission`` vs ``hasProjectPermission``? Slower request?